core/eolearn/core/utils/common.py

```diff
@@ -88,3 +88,34 @@ def generate_uid(prefix: str) -> str:
 def is_discrete_type(number_type: Union[np.dtype, type]) -> bool:
     """Checks if a given `numpy` type is a discrete numerical type."""
     return np.issubdtype(number_type, np.integer) or np.issubdtype(number_type, bool)
+
+
+def _apply_to_spatial_axes(
+    function: Callable[[np.ndarray], np.ndarray], data: np.ndarray, spatial_axes: Tuple[int, int]
+) -> np.ndarray:
+    """Helper function for applying a 2D -> 2D function to a higher dimension array
+
+    Recursively slices data into smaller-dimensional ones, until only the spatial axes remain. The indices of spatial
+    axes have to be adjusted if the recursion-axis is smaller than either one, e.g. spatial axes (1, 2) become (0, 1)
+    after splitting the 3D data along axis 0 into 2D arrays.
+
+    After achieving 2D data slices the mapping function is applied. The data is then reconstructed into original form.
+    """
+
+    ax1, ax2 = spatial_axes
+    if ax1 >= ax2:
+        raise ValueError("Values in `spatial_axes` should be increasing.")
+
+    if ax2 >= data.ndim:
+        raise ValueError("Values in `spatial_axes` should smaller then `data.ndim`.")
```

Comment:
```suggestion
        raise ValueError(f"Values in `spatial_axes` must be smaller than `data.ndim`, got {spatial_axes} for data of dimension {data.ndim}.")
```

Black will surely complain about this one :see_no_evil: